### DISCUSSION TITLE
Metasploit framework load error

### DISCUSSION THREAD

**Discussion by UnixStranger** [rapid7/metasploit-framework#16060] [Q&A] [answered]
Whenever I tried to launch Metasploit by using command: msfconsole
It's showing me this  Load error..
Anyone knows what type of error this is?

/data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:35:in `require': cannot load such file -- net/smtp (LoadError)
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:35:in `require'
        from /data/data/com.termux/files/home/metasploit-framework/lib/rex/proto/mms/model.rb:21:in `<top (required)>'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:27:in `require'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:27:in `require'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader/helpers.rb:95:in `const_get'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader/helpers.rb:95:in `cget'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader.rb:231:in `block (2 levels) in eager_load'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader/helpers.rb:26:in `block in ls'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader/helpers.rb:18:in `each_child'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader/helpers.rb:18:in `ls'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader.rb:226:in `block in eager_load'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader.rb:211:in `synchronize'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader.rb:211:in `eager_load'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader.rb:311:in `each'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/loader.rb:311:in `eager_load_all'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/railties-6.1.4.4/lib/rails/application/finisher.rb:133:in `block in <module:Finisher>'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/railties-6.1.4.4/lib/rails/initializable.rb:32:in `instance_exec'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/railties-6.1.4.4/lib/rails/initializable.rb:32:in `run'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/railties-6.1.4.4/lib/rails/initializable.rb:61:in `block in run_initializers'
        from /data/data/com.termux/files/usr/lib/ruby/3.1.0/tsort.rb:228:in `block in tsort_each'
        from /data/data/com.termux/files/usr/lib/ruby/3.1.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
        from /data/data/com.termux/files/usr/lib/ruby/3.1.0/tsort.rb:431:in `each_strongly_connected_component_from'
        from /data/data/com.termux/files/usr/lib/ruby/3.1.0/tsort.rb:349:in `block in each_strongly_connected_component'
        from /data/data/com.termux/files/usr/lib/ruby/3.1.0/tsort.rb:347:in `each'
        from /data/data/com.termux/files/usr/lib/ruby/3.1.0/tsort.rb:347:in `call'
        from /data/data/com.termux/files/usr/lib/ruby/3.1.0/tsort.rb:347:in `each_strongly_connected_component'
        from /data/data/com.termux/files/usr/lib/ruby/3.1.0/tsort.rb:226:in `tsort_each'
        from /data/data/com.termux/files/usr/lib/ruby/3.1.0/tsort.rb:205:in `tsort_each'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/railties-6.1.4.4/lib/rails/initializable.rb:60:in `run_initializers'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/railties-6.1.4.4/lib/rails/application.rb:391:in `initialize!'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/railties-6.1.4.4/lib/rails/railtie.rb:207:in `public_send'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/railties-6.1.4.4/lib/rails/railtie.rb:207:in `method_missing'
        from /data/data/com.termux/files/home/metasploit-framework/config/environment.rb:4:in `<top (required)>'
        from /data/data/com.termux/files/home/metasploit-framework/lib/msfenv.rb:17:in `require'
        from /data/data/com.termux/files/home/metasploit-framework/lib/msfenv.rb:17:in `<top (required)>'
        from /data/data/com.termux/files/usr/bin/msfconsole:18:in `require'
        from /data/data/com.termux/files/usr/bin/msfconsole:18:in `<main>'
/data/data/com.termux/files/usr/bin/msfconsole: warning: Exception in finalizer #<Proc:0x00000072c48c6e78 /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/concurrent-ruby-1.0.5/lib/concurrent/atomic/ruby_thread_local_var.rb:85>
/data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/logging-2.3.0/lib/logging/diagnostic_context.rb:471:in `new': can't alloc thread (ThreadError)
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/logging-2.3.0/lib/logging/diagnostic_context.rb:471:in `create_with_logging_context'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/logging-2.3.0/lib/logging/diagnostic_context.rb:436:in `new'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/concurrent-ruby-1.0.5/lib/concurrent/atomic/ruby_thread_local_var.rb:86:in `block in threadlocal_finalizer'
/data/data/com.termux/files/usr/bin/msfconsole: warning: Exception in finalizer #<Proc:0x00000072c48c6fe0 /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/concurrent-ruby-1.0.5/lib/concurrent/atomic/ruby_thread_local_var.rb:85>
/data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/logging-2.3.0/lib/logging/diagnostic_context.rb:471:in `new': can't alloc thread (ThreadError)
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/logging-2.3.0/lib/logging/diagnostic_context.rb:471:in `create_with_logging_context'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/logging-2.3.0/lib/logging/diagnostic_context.rb:436:in `new'
        from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/concurrent-ruby-1.0.5/lib/concurrent/atomic/ruby_thread_local_var.rb:86:in `block in threadlocal_finalizer'

**Comment by bcoles** [comment, marked as answer]
It looks like you're using Termux. Termux is not officially supported. Metasploit installation on Termux is not supported. Termux does not support hacking tools. #11023

> Anyone knows what type of error this is?
> /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:35:in require': cannot load such file -- net/smtp (LoadError) from /data/data/com.termux/files/usr/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:35:in require'

Ruby gem 'net-smtp' is missing.
